### DISCUSSION TITLE
Support for Official Logs Panel?

### DISCUSSION THREAD

**Discussion by risoms** [grafana/grafana-infinity-datasource#190] [Q&A] [answered]
Hello all. I'm interested in seeing if there's a way to use this wonderful plugin with the Logs Plugin? ([Example](https://play.grafana.org/d/6NmftOxZz/logs-panel?orgId=1&editPanel=2), [Description](https://grafana.com/docs/grafana/latest/visualizations/logs-panel/))

**Comment by yesoreyeram** [comment, marked as answer]
As long as there is one time column and minimum of one string column in your data that can be rendered in a logs panel. What problem you are facing? You can have a level column to indidicate the log level.

**Reply by risoms** [reply]
I was able to get it working eventually. My fault for not checking the refresh :P

**Comment by risoms** [comment]
@yesoreyeram Just a follow up on this. It looks like the logs panel indiscriminately provides row content. Is there a way to ensure each row is displaying intended content. Note: The detected fields dropdown seems to be working though.
![image](https://user-images.githubusercontent.com/10383917/142240425-d64520aa-2975-4316-9605-37ea54750d84.png)

**Reply by yesoreyeram** [reply]
That's the nature of log panel by design. First string column will always considered as message and the remaining columns will be considered fields. ( also there is a special column called level ) Read more [here](https://grafana.com/docs/grafana/latest/visualizations/logs-panel/)

**Reply by risoms** [reply]
@yesoreyeram How is the the first string column identified in a JSON-like structure that's being provided by grafana-infinity-datasource? Is it alphabetical? For example, the following data:
```json
[{
    "time": "2021-12-31 05:00",
    "something": "something else"
    "message": "hello"
},{
    "time": "2021-12-31 06:00",
    "something": "something else"
    "message": "hello"
}]
```

**Reply by yesoreyeram** [reply]
The order you defined in the "columns" of infinity. if nothing defined, grafana will try to define order from the JSON response.
